pandas/core/strings.py

```diff
@@ -45,6 +45,15 @@ def str_cat(arr, others=None, sep=None, na_rep=None):
 
     Examples
     --------
+    When ``na_rep`` is `None` (default behavior), NaN value(s)
+    in the Series propagate and return value will be NaN.
+
+    >>> Series(['a','b',np.nan,'c']).str.cat(sep=' ')
```

Comment:
I think ``.dropna`` (which doesn't exist on ``Index``) - its pretty easy (there is an issue about it).

so you can emulate so it will just work (for Series or Index)

``si.take(pd.notnull(si))``